ui/desktop/app/controllers/scopes.js

```diff
@@ -0,0 +1,64 @@
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { loading } from 'ember-loading';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+
+export default class ScopesController extends Controller {
+  // =services
+
+  @service ipc;
+  @service intl;
+  @service flashMessages;
+
+  // =attributes
+
+  metaData;
+  url;
+  @tracked error = false;
+
+  // =methods
+
+  @action
+  @loading
+  async downloadAndInstallCompatiableVersion() {
+    const metaDataUrl =
+      'https://api.releases.hashicorp.com/v1/releases/boundary-desktop/1.7.1';
+    const { isWindows, isMac, isLinux } = await this.ipc.invoke('checkOS');
+
+    try {
+      const metaDataResponse = await fetch(metaDataUrl);
+      this.metaData = await metaDataResponse.json();
+
+      if (isWindows) {
+        this.extractOsSpecificUrl('windows');
+      } else if (isMac) {
+        this.extractOsSpecificUrl('darwin');
+      } else if (isLinux) {
+        this.extractOsSpecificUrl('linux');
+      }
+
+      const response = await fetch(this.url);
+      const blob = await response.blob();
+
+      const urlParts = this.url.split('/');
+      const fileName = urlParts[urlParts.length - 1];
+
+      // Create an anchor element to trigger the download
+      const link = document.createElement('a');
+      link.href = window.URL.createObjectURL(blob);
+      link.download = fileName;
+
+      // Simulate click to trigger download
+      link.click();
```

Comment:
Is this just to have the download happen in the desktop client for a better UX? Is there a reason we couldn't just have the button link to the release url and it opens in the user's browser?

Reply:
Let me explore that change. That would be more what I would expect as a user.

Reply:
This process has been updated.